src/plugins/controls/public/options_list/embeddable/options_list_embeddable_factory.tsx

```diff
@@ -48,8 +48,11 @@ export class OptionsListEmbeddableFactory
       ((newInput.fieldName && !deepEqual(newInput.fieldName, embeddable.getInput().fieldName)) ||
         (newInput.dataViewId && !deepEqual(newInput.dataViewId, embeddable.getInput().dataViewId)))
     ) {
-      // if the field name or data view id has changed in this editing session, selected options are invalid, so reset them.
-      newInput.selectedOptions = [];
+      // if the field name or data view id has changed in this editing session, reset all selections
+      newInput.selectedOptions = undefined;
+      newInput.existsSelected = undefined;
+      newInput.exclude = undefined;
+      newInput.sort = undefined;
```

Comment:
Before these were added, `existsSelected`, `exclude`, and `sort` were all remembered when the data view and/or field of a control was changed - while it doesn't cause any apparent issues (unlike `selectedOptions`, which was the only thing included before), it goes against expected behaviour IMO. If the data view and/or field of a control is changed, I would expect the **entire** state of the control to be reset.